### PR TITLE
Implement Client::pull_manifest_raw

### LINKED ISSUE #114: Retrieving "unmangled" manifests for digest consistency?
Is there a way to fetch raw manifests without deserializing them immediately?

When retrieving manifests, they're automatically deserialized into Rust structures which is very handy, but unfortunately this also means that if you intend to push the manifest elsewhere your re-serialized version of the content might not match the digest provided by the upstream.

~~This problem can be worked around by simply not providing an expected digest when pushing to containerd for example, but this feels like a bit of a hack, since the changes to the underlying manifest (if any) are purely cosmetic, and the content and digest are in fact inconsistent.~~ Removing the expected digest simply allows the push to go through, but the final object will still have a different digest.

### PR DESCRIPTION
**Pull request by MathiasPius**
Implements `Client::pull_manifest_raw` for fetching the plain text representation of a manifest without deserializing it into an `OciManifest`.

This way the manifest can be pushed elsewhere while maintaining the digest value of the manifest which is otherwise subject to change following re-serialization of the `OciManifest` object.

### Asymmetry with `push_raw_manifest`
`push_manifest_raw` takes a `Vec<u8>`, while `pull_manifest_raw` returns the content as a `String`. I've chosen this route, since it allows us to still call `validate_registry_response` in `Client::_pull_manifest_raw` where the response status and manifest text is available for validation.

### Providing `accepted_media_types` when pulling raw manifests
Since `pull_manifest_raw` is specifically intended for fetching individual manifests for mirroring, I expect the caller to be traversing the tree of manifests manually and pushing/storing them elsewhere, and in that case they will have a very good idea of what kind of manifest they're likely to get back at each individual step.

### Deduplicating code
I've re-implemented `Client::_pull_manifest` in terms of `Client::_pull_manifest_raw` to reduce code duplication, so it now uses the raw function to get the text manifest, and then performs deserialization from there.

Closes #114

### CODE REVIEW

File: src/client.rs
Comment (thomastaylor312) [inline src/client.rs:746]: I still think I'd prefer to have the symmetry of this being a `Vec<u8>` because I think that is what would be expected when pulling something raw. I totally understand the motivation behind keeping it a string though. I think the validate function should just be updated to take `&[u8]`, but I don't want to make you do that if you don't want to. So if you'd like to do it, let me know. Otherwise, I can merge this after a day or so if I haven't heard from you and make the change in a follow up PR!

**Review by thomastaylor312 — APPROVED** [review]
Waiting for a day to merge per my review comment. Thanks for adding this @MathiasPius!

File: src/client.rs
Comment (flavio) [reply src/client.rs:746]: I agree with @thomastaylor312 on this request

File: src/client.rs
Comment (flavio) [inline src/client.rs:2757]: This test is not making any request against the local registry. You can drop the feature `test-restry` and the code that deal with starting the local registry

**Review by flavio — CHANGES_REQUESTED** [review]

File: src/client.rs
Comment (MathiasPius) [reply src/client.rs:2757]: Ah correct! In a previous iteration I did have it use the local registry, but it didn't really add any assurances to the test.

File: src/client.rs
Comment (MathiasPius) [reply src/client.rs:746]: Perfectly reasonable, I'll see what I can do!

**Comment by MathiasPius** [comment]
Thanks for the quick reviews! I've implemented the changes as requested, please review the updated PR, and let me know if there's anything else :)

**Comment by flavio** [comment]
@MathiasPius can you please fix the small formatting issue that causes the windows test to fail?

Aside from that, the PR LGTM

**Review by flavio — APPROVED** [review]
Thanks for having addresses all the issues

**Pull request merged**
